### ISSUE TITLE
VMA ERROR: qpm[0x2361a80]:625:prepare_ibv_qp() failed to modify QP from ERR to INIT state (ret = -3)

### ISSUE THREAD

**Issue by sm1ly** [Mellanox/libvma#691]
Using:
centos 7 with:
- 3.10.0-862.6.3.rt56.811.el7.x86_64
- Mellanox ConnectX InfiniBand driver v4.4-1.0.0
- VMA_VERSION: 8.6.10-0

and command:
sockperf sr -i 10.10.10.10

cant use network...

**Comment by liranoz12** [comment]
Hi @sm1ly,

It seems to be a failure in _ibv_modify_qp_/_ibv_exp_modify_qp_.
Are you able to run other verbs applications such as _ibv_rc_pingpong_ ? 
Please make sure there are no leftovers from previous libibverbs installation (verbs.h/ verbs_exp.h libibverbs.so etc.).

Thanks,
Liran

**Comment by liranoz12** [comment]
@sm1ly, 
Do you use bonding at your setup ? If yes, with which adapter? 
it seems related to RoCE LAG (see release notes for more info + workaround).
Please try to reprodcue the issue using VMA 8.7.5 - you sould see a proper warning if RoCE lag is enabled.

Thanks,
Liran

**Comment by sm1ly** [comment]
Hi @liranoz12 ,
Yes we got bonding with vlans. And we use it with mellanox connectx3.
We testing now with 3.10.0-862.14.4.rt56.821 kernel and with OFED 4.4-2.0.7.0.
And with vma 8.7.5
So I got warnings like 

> VMA cannot offload the device while RoCE LAG is enabled.

but we using mlx4_core. so I havent got roce_lag_enable in system absolutely.
And your workaround doesnt work.

But I got some strange info. We got another host with different HW and mellanox connectx3 (the same)
with the same drivers and kernel. and test with vma 8.0.4 and 8.7.5 and have no errors at all.
IDK what influence on it.

**Comment by liranoz12** [comment]
Hi @sm1ly,

I tested the workaround again (as described in the[ User Manual page 73](http://www.mellanox.com/related-docs/prod_acceleration_software/VMA_8_7_5_User_Manual.pdf)) and it seems to work.
Cloud you please try it again ? 

[root@r-aa-zorro006 libvma]# uname -r && cat /etc/redhat-release && ofed_info -s
3.10.0-514.el7.x86_64
Red Hat Enterprise Linux Server release 7.3 (Maipo)
MLNX_OFED_LINUX-4.5-1.0.1.0:

[root@r-aa-zorro006 libvma]# cat /proc/net/bonding/bond0
Ethernet Channel Bonding Driver: v3.7.1 (April 27, 2011)

Bonding Mode: fault-tolerance (active-backup)
Primary Slave: None
Currently Active Slave: ens1
MII Status: up
MII Polling Interval (ms): 100
Up Delay (ms): 0
Down Delay (ms): 0

Slave Interface: ens1
MII Status: up
Speed: 40000 Mbps
Duplex: full
Link Failure Count: 0
Permanent HW addr: 00:02:c9:34:1d:f0
Slave queue ID: 0

Slave Interface: ens1d1
MII Status: up
Speed: 40000 Mbps
Duplex: full
Link Failure Count: 0
Permanent HW addr: 00:02:c9:34:1d:f1
Slave queue ID: 0

[root@r-aa-zorro006 libvma]# ibdev2netdev
mlx4_0 port 1 ==> bond0 (Up)
mlx5_0 port 1 ==> ib0 (Down)
mlx5_1 port 1 ==> ens2f1 (Up)
mlx5_2 port 1 ==> ens4f0 (Up)
mlx5_3 port 1 ==> ens4f1 (Up)

[root@r-aa-zorro006 libvma]# LD_PRELOAD=libvma.so sockperf sr -i 1.1.1.1
 VMA INFO: ---------------------------------------------------------------------------
 VMA INFO: VMA_VERSION: 8.7.5-0 Release built on Nov 28 2018 12:34:43
 VMA INFO: Cmd Line: sockperf sr -i 1.1.1.1
 VMA INFO: OFED Version: MLNX_OFED_LINUX-4.5-1.0.1.0:
 VMA INFO: ---------------------------------------------------------------------------
 VMA INFO: Log Level                      INFO                       [VMA_TRACELEVEL]
 VMA INFO: ---------------------------------------------------------------------------
 VMA WARNING: *******************************************************************************************************
 VMA WARNING: * Interface bond0 will not be offloaded.
 VMA WARNING: * VMA cannot offload the device while RoCE LAG is enabled.
 VMA WARNING: * Please refer to VMA Release Notes for more info
 VMA WARNING: ******************************************************************************************************
 VMA WARNING: *******************************************************************************************************
 VMA WARNING: * Bond bond0 will not be offloaded due to problem with its slaves.
 VMA WARNING: * Check warning messages for more information.
 VMA WARNING: *******************************************************************************************************
sockperf: == version #3.6-no.git ==
sockperf: [SERVER] listen on:
[ 0] IP = 1.1.1.1         PORT = 11111 # UDP
sockperf: Warmup stage (sending a few dummy messages)...
sockperf: [tid 16809] using recvfrom() to block on socket(s)
^Csockperf: Test end (interrupted by user)
sockperf: No messages were received on the server.
sockperf: cleanupAfterLoop() exit

[root@r-aa-zorro006 libvma]# ip link add dummy0 type dummy
[root@r-aa-zorro006 libvma]#  ifenslave -d bond0 ens1 ens1d1
[root@r-aa-zorro006 libvma]# ifenslave bond0 dummy0 ens1 ens1d1
[root@r-aa-zorro006 libvma]#  ifenslave -d bond0 dummy0
[root@r-aa-zorro006 libvma]# ip link del dummy0 type dummy

[root@r-aa-zorro006 libvma]# ibdev2netdev
mlx4_0 port 1 ==> bond0 (Up)
mlx4_0 port 2 ==> bond0 (Up)
mlx5_0 port 1 ==> ib0 (Down)
mlx5_1 port 1 ==> ens2f1 (Up)
mlx5_2 port 1 ==> ens4f0 (Up)
mlx5_3 port 1 ==> ens4f1 (Up)

[root@r-aa-zorro006 libvma]# cat /proc/net/bonding/bond0
Ethernet Channel Bonding Driver: v3.7.1 (April 27, 2011)

Bonding Mode: fault-tolerance (active-backup)
Primary Slave: None
Currently Active Slave: ens1
MII Status: up
MII Polling Interval (ms): 100
Up Delay (ms): 0
Down Delay (ms): 0

Slave Interface: ens1
MII Status: up
Speed: 40000 Mbps
Duplex: full
Link Failure Count: 0
Permanent HW addr: 00:02:c9:34:1d:f0
Slave queue ID: 0

Slave Interface: ens1d1
MII Status: up
Speed: 40000 Mbps
Duplex: full
Link Failure Count: 0
Permanent HW addr: 00:02:c9:34:1d:f1
Slave queue ID: 0

[root@r-aa-zorro006 libvma]# LD_PRELOAD=libvma.so sockperf sr -i 1.1.1.1
 VMA INFO: ---------------------------------------------------------------------------
 VMA INFO: VMA_VERSION: 8.7.5-0 Release built on Nov 28 2018 12:34:43
 VMA INFO: Cmd Line: sockperf sr -i 1.1.1.1
 VMA INFO: OFED Version: MLNX_OFED_LINUX-4.5-1.0.1.0:
 VMA INFO: ---------------------------------------------------------------------------
 VMA INFO: Log Level                      INFO                       [VMA_TRACELEVEL]
 VMA INFO: ---------------------------------------------------------------------------
sockperf: == version #3.6-no.git ==
sockperf: [SERVER] listen on:
[ 0] IP = 1.1.1.1         PORT = 11111 # UDP
sockperf: Warmup stage (sending a few dummy messages)...
sockperf: [tid 16952] using recvfrom() to block on socket(s)
^Csockperf: Test end (interrupted by user)
sockperf: No messages were received on the server.
sockperf: cleanupAfterLoop() exit

Thanks,
Liran

**Comment by sm1ly** [comment]
Hello Liran.
So I got working workaround with dummy. But I got another error now:
[root@bcs2 ~]# sockperf sr --load-vma -i 10.10.10.10
 VMA INFO: ---------------------------------------------------------------------------
 VMA INFO: VMA_VERSION: 8.7.5-0 Development Snapshot built on Dec 10 2018 22:35:34
 VMA INFO: Git: a0c576b7f08e099a7579ef61ec2888962bff0837+
 VMA INFO: Cmd Line: sockperf sr --load-vma -i 10.10.10.10
 VMA INFO: Current Time: Mon Dec 10 22:48:08 2018
 VMA INFO: Pid: 20493
 VMA INFO: OFED Version: MLNX_OFED_LINUX-4.5-1.0.1.0:
 VMA INFO: Architecture: x86_64
 VMA INFO: Node: bcs2
 VMA INFO: ---------------------------------------------------------------------------
 VMA INFO: Log Level                      INFO                       [VMA_TRACELEVEL]
 VMA INFO: ---------------------------------------------------------------------------
sockperf: symbol lookup error: /lib64/libvma.so: undefined symbol: _ZN20net_device_table_mgr11get_ip_listEi
[root@bcs2 ~]# sockperf | head -n2
sockperf is a tool for testing network latency and throughput.
version 3.6-no.git
